README.md

```diff
@@ -7,42 +7,87 @@
 
 
 ## About
-This is a simple REST client for C++. It wraps libcurl for HTTP requests.
+This is a simple REST client for C++. It wraps [libcurl][] for HTTP requests.
 
 ## Usage
-I tried to keep usage close to the [ruby rest-client][]. So the basic usage is:
+restclient-cpp provides two ways of interacting with REST endpoints. There is
+a simple one, which doesn't need you to configure an object to interact with
+an API. However the simple way doesn't provide a lot of configuration options
+either. So if you need more than just a simple HTTP call, you will probably
+want to check out the advanced usage.
 
-```cpp
-RestClient::method(url, content-type, params);
-// or
-RestClient::method(url, content-type, params, headers);
-```
-
-Examples:
+### Simple API
+The simple API is just some static methods modeled after the most common HTTP
+verbs:
 
 ```cpp
 #include "restclient-cpp/restclient.h"
 
+// initialize RestClient
+RestClient::init()
+
 RestClient::response r = RestClient::get("http://url.com")
 RestClient::response r = RestClient::post("http://url.com/post", "text/json", "{\"foo\": \"bla\"}")
 RestClient::response r = RestClient::put("http://url.com/put", "text/json", "{\"foo\": \"bla\"}")
 RestClient::response r = RestClient::del("http://url.com/delete")
 
-// add some headers
+// deinit RestClient. After calling this you have to call RestClient::init()
+// again before you can use it
+RestClient::disable();
+```
+
+The response is of type [RestClient::response][restclient_response] and has
+three attributes:
+
+```cpp
+RestClient::response.code // HTTP response code
+RestClient::response.body // HTTP response body
+RestClient::response.headers // HTTP response headers
+```
+
+### Advanced Usage
+However if you want more sophisticated features like connection reuse,
+timeouts or authentication, there is also a different, more configurable way.
 
+```cpp
+#include "restclient-cpp/restclient.h"
+
+// initialize RestClient
+RestClient::init();
+
+// get a connection object
+RestClient::Connection conn = new RestClient::Connection();
```

Comment:
oh that's a good idea. Did you have other advantages in mind that this would enable besides object lifecyle management?